README.md

```diff
@@ -0,0 +1,182 @@
+
+# Building
+## Requirements
+
+1. Ubuntu 18.0+ or any other Linux distribution which is supported by Poky/OE
+2. Development packages for Yocto. Refer to [Yocto
+   manual](https://www.yoctoproject.org/docs/current/mega-manual/mega-manual.html#brief-build-system-packages).
+3. You need `Moulin` installed in your PC. Recommended way is to
+   install it for your user only: `pip3 install --user
+   git+https://github.com/xen-troops/moulin`. Make sure that your
+   `PATH` environment variable includes `${HOME}/.local/bin`.
+4. Ninja build system: `sudo apt install ninja-build` on Ubuntu
+
+## Fetching
+
+You can fetch/clone this whole repository, but you actually only need
+one file from it: `prod-devel-rcar-s4.yaml`. During the build `moulin`
+will fetch this repository again into `yocto/` directory. So, to
+reduce possible confuse, we recommend to download only
+`prod-devel-rcar-s4.yaml`:
+
+```
+# curl -O https://raw.githubusercontent.com/xen-troops/meta-xt-prod-devel-rcar-gen4/spider-0.8.2-l3-offload/prod-devel-rcar-s4.yaml
+```
+
+## Building
+
+Moulin is used to generate Ninja build file: `moulin
+prod-devel-rcar-s4.yaml`.
+
+## Creating SD card image
+
+### Using Ninja
+
+Latest versions of moulin will generate additional ninja targets for creating images. In case of this product please use
+
+```
+# ninja image-full
+```
+
+To generate SD/eMMC image `full.img`. You can use `dd` tool to write
+this image file to a SD card:
+
+```
+# dd if=full.img of=/dev/sdX conv=sparse
+```
+
+If you want to write image directly to a SD card (e.g. without
+creating `full.img` file), you will need to use manual mode, which is
+described in the next subsection.
+
+### Manually, using `rouge`
+
+Image file can be created with `rouge` tool. This is a companion
+application for `moulin`.
+
+Right now it works only in standalone mode, so manual invocation is
+required.
+
+This XT product provides only one image: `full`.
+
+You can prepare the image by running
+
+```
+# rouge prod-devel-rcar-s4.yaml -i full
+```
+
+This will create file `full.img` in your current directory.
+
+Also you can write image directly to an SD card by running
+
+```
+# sudo rouge prod-devel-rcar.yaml -i full -so /dev/sdX
+```
+
+**BE SURE TO PROVIDE CORRECT DEVICE NAME**. `rouge` has no
+interactive prompts and will overwrite your device right away. **ALL
+DATA WILL BE LOST**.
+
+It is also possible to flash the image to the internal eMMC.
+For that you may want booting the board via TFTP and sharing DomD's
+root file system via NFS. Once booted it is possible to flash the image:
+
+```
+# dd if=/full.img of=/dev/mmcblk0 bs=1M
+```
+
+For more information about `rouge` check its
+[manual](https://moulin.readthedocs.io/en/latest/rouge.html).
+
+## U-boot environment
+
+It is possible to run the build from TFTP+NFS or eMMC. With NFS boot
+it is possible to configure board IP, server IP and NFS path for DomD.
```

Comment:
We have no DomD there. Call it just "system"

Reply:
Renamed. Also, renamed DomD naming in another place